scripts/extensions/videoEditor/src/VideoEditorThumbnail.tsx

```diff
@@ -0,0 +1,361 @@
+import * as React from 'react';
+import {ISuperdesk, IArticle} from 'superdesk-api';
+import VideoEditorContext from './VideoEditorContext';
+import {IVideoEditor} from './interfaces';
+import {pick} from 'lodash';
+
+interface IProps {
+    videoRef: React.RefObject<HTMLVideoElement>;
+    article: IArticle;
+    crop: IVideoEditor['crop'];
+    rotate: IVideoEditor['degree'];
+    onToggleLoading: (isLoading: boolean, loadingText?: string) => void;
+    onSave: (article: IArticle) => void;
+    onError: (err: any) => void;
+    getCropRotate: (crop: IVideoEditor['crop']) => IVideoEditor['crop'];
+}
+
+interface IState {
+    dirty: boolean;
+    type: 'capture' | 'upload' | '';
+    value: number | File; // capture positon or uploaded File
+    // most recent updated thumbnail, for re-drawing thumbnail when reset changes
+    thumbnail: string;
+    // save current rotate degree when user captures thumbnail
+    rotateDegree: number;
+    scale: number;
+}
+
+export class VideoEditorThumbnail extends React.Component<IProps, IState> {
+    static contextType = VideoEditorContext;
+    private ref: React.RefObject<HTMLCanvasElement>;
+    private maxCanvasSize: { width: number; height: number };
+    private interval: number;
+
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            dirty: false,
+            type: '',
+            value: 0,
+            thumbnail: this.props.article.renditions?.thumbnail!?.href,
+            rotateDegree: 0,
+            scale: 1,
+        };
+        this.ref = React.createRef();
+        this.maxCanvasSize = {width: 0, height: 0};
+        this.interval = 0;
+    }
+
+    componentDidMount() {
+        if (this.state.thumbnail) {
+            const thumbnail = this.state.thumbnail + `?t=${Math.random()}`;
```

Comment:
It can, but `Cache-Control` needs to be `no-store` for it to work, this mean thumbnail will be re-loaded every time user click Cancel. 
While checking this, it seems `thumbnail` state in above comment is needed so that thumbnail will not need to reload on canceling. 
Should we leave this for server to handle?

Reply:
> It can, but Cache-Control needs to be no-store for it to work, this mean thumbnail will be re-loaded every time user click Cancel.

You should be able to apply different caching policies based on the endpoint. If it's hard to do, remove caching for all video editor endpoints. Thumbnails don't require much data and I don't think the use case of closing and reopening the same video would be that common.

Reply:
Sorry if not make clear, by reload thumbnail, we mean capturing a preview thumbnail then click cancel, if there is no caching in place, it will reload each time user click capture then cancel.
But we do agree, we should not worry about reloading thumbnail, if thumbnail size is too big, it's server responsibility to optimize it